app/gosqs/gosqs.go

```diff
@@ -335,9 +335,9 @@ func ReceiveMessage(w http.ResponseWriter, req *http.Request) {
 
 	loops := waitTimeSeconds * 10
 	for loops > 0 {
-		app.SyncQueues.Lock()
+		app.SyncQueues.RLock()
```

Comment:
Now that I look at this I do not think we need a Lock/unlock around that line at all.  Just remove it.

Reply:
But in case of removal of locks here we still keep locks in `numberOfHiddenMessagesInQueue` function then?

Reply:
@p4tin ping

Also wouldn't you like consistent app.SyncQueues.Queues[queueName] between `len` and `numberOfHiddenMessagesInQueue` calls? Having a lock just inside numberOfHiddenMessagesInQueue wouldn't guarantee that

Reply:
you are correct